src/features/unified-share-modal/ContactsField.js

```diff
@@ -90,22 +81,21 @@ class ContactsField extends React.Component<Props, State> {
                 .filter(
                     // filter contacts who have already been selected
                     ({ email, id }) => !selectedContacts.find(({ value }) => value === email || value === id),
-                )
-                .map<Object>(({ email, id, isExternalUser, name, type }) => ({
-                    // map to standardized DatalistItem format
-                    email,
-                    id,
-                    isExternalUser,
-                    text: name,
-                    type,
-                    value: email || id, // if email doesn't exist, contact is a group, use id
-                }));
+                );
 
             if (suggestedCollaborators) {
-                return this.addSuggestedContacts(fullContacts);
+                fullContacts = this.addSuggestedContacts(fullContacts);
```

Comment:
Slightly prefer setting this new list into a separate variable, then merging it below, and mapping over the result, so it's more clear (and we can avoid having a variable that mutates).

Unless `.addSuggestedContacts` is a superset of what `fullContacts` is before this runs...

Reply:
The problem is that addSuggestedContacts not only adds contacts to the front but also removes contacts from the original list, so there's not really a way of avoiding mutation without duplicating code